Document delta saving feature.

mattstein opened the pull request:
I've attempted to describe the field and element type changes that can be made to implement delta saving. Hopefully without excessive verbosity.

mattstein committed:
Document delta saving feature.

mattstein committed:
Swap blurbs, add reference to Element Types.

mattstein committed:
Fix case.

mattstein committed:
Clarify context.

mattstein committed:
Avoid tragic widow.

mattstein committed:
Unproperize.

mattstein committed:
Update element type content delta notes.

- Re-title section.
- Element type, not field type.
- Used sentence case for numbered points.
- Removed incorrect `view.registerDeltaName()` note and instead noted `registerDeltas: true` for field includes.
- Added combined+highlighted template update example.

mattstein committed:
Update field type delta save section.

- Re-titled.
- Adjusted wording, linked methods to API docs.
- Removed last sentence.

brandonkelly commented on extend/field-types.md:
```diff
@@ -32,3 +32,22 @@ class Plugin extends \craft\base\Plugin
+
+## Supporting Delta Saves
+
+If your field type needs to do anything following [`afterElementSave()`](api:craft\base\Field::afterElementSave()) or [`afterElementPropagate()`](api:craft\base\Field::afterElementPropagate()), you can improve performance by skipping processing when the field’s value is unchanged.
```
```suggestion
If your field type does any processing from [afterElementSave()](api:craft\base\FieldInterface::afterElementSave()) or [afterElementPropagate()](api:craft\base\FieldInterface::afterElementPropagate()), you can improve performance by skipping processing when the field’s value is unchanged.
```

brandonkelly commented on extend/element-types.md:
````diff
@@ -857,3 +857,24 @@ public function setEagerLoadedElements(string $handle, array $elements)
+
+```twig{1,6}
+{% do view.setIsDeltaRegistrationActive(true) %}
+
````
Maybe put a `...` line here surrounded by blank lines, so it’s more clear that these wouldn’t go right next to each other. (Don’t forget to change `{1,6}` to `{1,8}` in the process.)

brandonkelly commented on extend/field-types.md:
```diff
@@ -32,3 +32,22 @@ class Plugin extends \craft\base\Plugin
+
+If your field type needs to do anything following [`afterElementSave()`](api:craft\base\Field::afterElementSave()) or [`afterElementPropagate()`](api:craft\base\Field::afterElementPropagate()), you can improve performance by skipping processing when the field’s value is unchanged.
+
+You can tell whether field content has changed by calling `isFieldDirty()` on the element.
```
```suggestion
You can tell whether field content has changed by calling [isFieldDirty()](api:craft\base\ElementInterface::isFieldDirty()) on the element.
```

brandonkelly commented on extend/field-types.md:
```diff
@@ -32,3 +32,22 @@ class Plugin extends \craft\base\Plugin
+
+You can tell whether field content has changed by calling `isFieldDirty()` on the element.
+
+This method checks for content changes before continuing to handle that element:
```
This line can probably be removed, as it’s not really adding anything.

mattstein committed:
Improve wording.

Co-Authored-By: Brandon Kelly <brandon@pixelandtonic.com>

mattstein committed:
Hint that lines don’t go next to each other.

mattstein committed:
Link `isFieldDirty()`.

Co-Authored-By: Brandon Kelly <brandon@pixelandtonic.com>

mattstein committed:
Strip cruft.

mattstein reviewed (commented):
All updated. Thanks @brandonkelly!

mattstein committed:
Update Testing section.

Add `Testing` prefix to some titles to help differentiate search results, clean up some minor typos, wording, and formatting issues.
